### ISSUE TITLE
Issues in installing papaja

### ISSUE THREAD

**Issue by koliii** [crsh/papaja#290]
R version 3.4.4 (2018-03-15) -- "Someone to Lean On"

Platform: x86_64-pc-linux-gnu (64-bit)

I assume below output is not normal and something is wrong while installing papaja.
`
The downloaded source packages are in
	‘/tmp/RtmpyYt2CK/downloaded_packages’
✔  checking for file ‘/tmp/RtmpyYt2CK/remotes19e54a9b9aef/crsh-papaja-b6cd70f/DESCRIPTION’ ...
─  preparing ‘papaja’:
✔  checking DESCRIPTION meta-information
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘papaja_0.1.0.9842.tar.gz’
   
Installing package into ‘/home/ajay/R/x86_64-pc-linux-gnu-library/3.4’
(as ‘lib’ is unspecified)
* installing *source* package ‘papaja’ ...
** R
** inst
** preparing package for lazy loading
in method for ‘apa_print’ with signature ‘"BFBayesFactor"’: no definition for class “BFBayesFactor”
in method for ‘apa_print’ with signature ‘"BFBayesFactorTop"’: no definition for class “BFBayesFactorTop”
in method for ‘apa_print’ with signature ‘"BFBayesFactorList"’: no definition for class “BFBayesFactorList”
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (papaja)
Warning message:
In is.na(remote_deps$package) :
  is.na() applied to non-(list or vector) of type 'NULL'
`

**Comment by koliii** [comment]
This is the error on the running the provide apa template:


tlmgr search --file --global '/apa6.cls'

tlmgr: Remote repository is newer than local (2018 < 2019)
Cross release updates are only supported with
  update-tlmgr-latest(.sh/.exe) --update
Please see https://tug.org/texlive/upgrade.html for details.
! LaTeX Error: File `apa6.cls' not found.

! Emergency stop.
<read *> 

Error: Failed to compile ex.tex. See https://yihui.name/tinytex/r/#debugging for debugging tips. See ex.log for more info.
In addition: Warning messages:
1: In cite_r("r-references.bib") :
  File r-references.bib not found. Cannot cite R-packages. If knitting again does not solve the problem, please check file path.
2: running command ''tlmgr' search --file --global '/apa6.cls'' had status 1 
Execution halted

**Comment by crsh** [comment]
It looks like you are missing the `apa6` LaTeX package. Please make sure you have set up your TeX distribution [according to the manual](https://crsh.github.io/papaja_man/introduction.html#software-requirements).

**Comment by koliii** [comment]
Thanks, I thought that R package `tinytex` will take care of `apa6` also.

**Comment by crsh** [comment]
Indeed, it should. Sorry about that. You're version of `tinytex` is outdated, though. Have you tried reinstalling it?

**Comment by koliii** [comment]
`tinytex 0.13` is installed. But this issue is not solved. I tried to reintall `papaja`, below is the output:


> if(!"devtools" %in% rownames(installed.packages())) install.packages("devtools")
> 
> devtools::install_github("crsh/papaja")
Skipping install of 'papaja' from a github remote, the SHA1 (b6cd70f6) has not changed since last install.
  Use `force = TRUE` to force installation
> 
> devtools::install_github("crsh/papaja@devel")
Downloading GitHub repo crsh/papaja@devel
These packages have more recent versions available.
Which would you like to update?

1:   mime (0.6 -> 0.7) [CRAN]

Enter one or more numbers separated by spaces, or an empty line to cancel
1: 1
mime (0.6 -> 0.7) [CRAN]
Installing 1 packages: mime
Installing package into ‘/home/ajay/R/x86_64-pc-linux-gnu-library/3.4’
(as ‘lib’ is unspecified)
trying URL 'https://cloud.r-project.org/src/contrib/mime_0.7.tar.gz'
Content type 'application/x-gzip' length 13130 bytes (12 KB)
==================================================
downloaded 12 KB

* installing *source* package ‘mime’ ...
** package ‘mime’ successfully unpacked and MD5 sums checked
** libs
gcc -std=gnu99 -I/usr/share/R/include -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-AitvI6/r-base-3.4.4=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c init.c -o init.o
gcc -std=gnu99 -I/usr/share/R/include -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-AitvI6/r-base-3.4.4=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c rawmatch.c -o rawmatch.o
g++ -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o mime.so init.o rawmatch.o -L/usr/lib/R/lib -lR
installing to /home/ajay/R/x86_64-pc-linux-gnu-library/3.4/mime/libs
** R
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (mime)

The downloaded source packages are in
	‘/tmp/RtmpjJrSSP/downloaded_packages’
✔  checking for file ‘/tmp/RtmpjJrSSP/remotes6c33df8046/crsh-papaja-ce13769/DESCRIPTION’ ...
─  preparing ‘papaja’:
✔  checking DESCRIPTION meta-information ...
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘papaja_0.1.0.9842.tar.gz’
   
Installing package into ‘/home/ajay/R/x86_64-pc-linux-gnu-library/3.4’
(as ‘lib’ is unspecified)
* installing *source* package ‘papaja’ ...
** R
** inst
** preparing package for lazy loading
in method for ‘apa_print’ with signature ‘"BFBayesFactor"’: no definition for class “BFBayesFactor”
in method for ‘apa_print’ with signature ‘"BFBayesFactorTop"’: no definition for class “BFBayesFactorTop”
in method for ‘apa_print’ with signature ‘"BFBayesFactorList"’: no definition for class “BFBayesFactorList”
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (papaja)

**Comment by crsh** [comment]
Sorry, I wasn't clear. What I meant is use `reinstall_tinytex()` to reinstall the `tinytex` TeX distribution.

**Comment by crsh** [comment]
Did this solve the issue for you?

**Comment by koliii** [comment]
No. Still showing the same error on knitting the document.

**Comment by crsh** [comment]
Hmm. I think this may be something specific to your TinyTeX setup (rather than `papaja`). It cannot install the `apa6` document class because your distribution is "outdated".  According to the [TinyTeX FAQ](https://yihui.name/tinytex/faq/) this error should be resolved by running reinstalling `tinytex` or running `tinytex::reinstall_tinytex()`. You could try `tinytex::reinstall_tinytex(force = TRUE)`. If that doesn't solve the problem, I'm not sure how to proceed. This may be something to ask the `tinytex`-developers.

**Comment by fred-oswald** [comment]
What is the latest version of R with which papaja is compatible? Having a hard time installing - thanks!

**Comment by crsh** [comment]
It should be compatible with all the newest versions. Have you looked at the [installation instructions](https://crsh.github.io/papaja_man/introduction.html#getting-started)? If you provide more context I'd be happy to look into it.

**Comment by fred-oswald** [comment]
got it working, thank you! :)
=
Fred Oswald
Website <http://workforce.rice.edu/> | Calendar
<http://www.tinyurl.com/foswald> | Twitter <https://twitter.com/FredOswald>
=
Pursue graduate study <https://graduate.rice.edu/> at Rice University, one
of America’s leading research institutions. Click here
<https://graduate.rice.edu/applicants/graduate-study-rice/interest-form> to
request information.
